Core/Parsing/ExpressionVisitors/TreeEvaluation/EvaluatableTreeFindingExpressionVisitor.cs

```diff
@@ -552,8 +555,10 @@ private bool IsCurrentExpressionEvaluatable (Expression expression)
         return IsCurrentExpressionEvaluatable (reducedExpression);
       }
 
-      // Note: In .NET 4.0, all expressions supplied by System.Core have a dedicated Visit method provided by System.Linq.Parsing.ExpressionVisitor.
-      return Enum.IsDefined (typeof (ExpressionType), expression.NodeType);
+      // PERF: We don't use Enum.IsDefined because it is too slow. We can hard-code ExpressionType.Add
+      // because it will never change. 
+      return expression.NodeType >= ExpressionType.Add
```

Comment:
Would it be better to place the ExpressionType.Add value into a static field similiar to the max-value and add the comment regarding the unchanging nature there. That way, the code becomes more symmetrical.

Also, please keep the previous note, too, since it explains a different detail.